mdazfar2/Ezyshop
# Setup store api and folder restructuring and rewriting logic of seller dashboard

## Description

# PR Description

## Overview
- Set up the backend endpoints to create a store for newly signed-up sellers.
- Ensured that after the store creation form submission, the seller is **redirected** to the respective store dashboard.
- Implemented **edit functionality** to allow sellers to modify store data post-creation.

## Changes Made
- **Backend**:
  - Added API endpoints to handle **store creation** for a new seller.
  - Ensured the store is linked to the seller correctly via their `sellerId`.
- **Frontend**:
  - Redirected sellers to their **dashboard** after successful form submission.
  - Implemented **store data editing** functionality with form pre-filling and validation.

## Testing
- Verified that:
  - A new store is correctly associated with the seller.
  - After form submission, the seller is redirected to their store dashboard.
  - Editing store details works, and updates are persisted in the backend.


## How to Test
1. Sign up as a seller and fill out the **store creation form**.
2. Submit the form and verify redirection to the store dashboard.
3. Navigate to the edit page and modify the store data.
4. Ensure that the changes reflect correctly in both the frontend and backend.

## Additional Notes
- Used **Prisma ORM** for database operations.
- Error handling and edge case management implemented (e.g., missing fields, invalid inputs).



<br/>



## Related Issues


- Closes #1032 


<br/>


## Screenshots / videos (if applicable)


https://github.com/user-attachments/assets/26af82d5-3df6-482b-bcc5-26ec61dc5d11


https://github.com/user-attachments/assets/5c2db9b8-7d1f-4916-9dff-4da29ff17655
